src/compute-client/src/controller.rs

```diff
@@ -730,8 +714,10 @@ where
         &mut self,
         instance_id: ComputeInstanceId,
         dataflow: DataflowDescription<mz_compute_types::plan::Plan<T>, (), T>,
+        subscribe_target_replica: Option<ReplicaId>,
```

Comment:
Should we switch the order of the parameters? `ship_dataflow` has `dataflow` first. I think that'd be nicer because instance and target replica wouldn't be split apart.

Reply:
That reasoning makes sense, but I have a counter-reasoning too: All methods of the compute controller that refer to an instance have the `instance_id` as the first parameter, so making it the second parameter here would make things less consistent.

We could make `subscribe_target_replica` the second parameter, which would be in line with both of the reasonings. But then we'd have an optional parameter that's `None` in most cases in a prominent position, not sure if that's better.